minio/minio
# remove unnecessary code checking for supported features

## Description
remove unnecessary code checking for supported features

Bonus: optimize markDelete/purgeDelete code for buckets 
replicating in site-replication to use peer calls.

## Motivation and Context
code cleanup

## How to test this PR?
Nothing should change or break, site-replication tests 
already provide coverage for the bucket delete sync
calls.

## Types of changes

## Checklist:
